rectools/metrics/classification.py

```diff
@@ -33,7 +34,7 @@
 
 
 @attr.s
-class ClassificationMetric(MetricAtK):
+class ClassificationMetric(DibiasableMetrikAtK):
```

Comment:
We need to add `DibiasableMetrikAtK` inheritange to `PartialAUC` and `PAP` metrics. We had a PR with them merged recently.

Reply:
I added `DibiasableMetrikAtK` inheritance from `PartialAUC` and `PAP`.
I have the following questions: Is it correct to apply debias method for `outer_merged_enriched` which is returned after `AUCMetric.fit(...)`? This is main point, since the calculation of `PartialAUC` and `PAP` metrics actually happens in the `calc_per_user_from_fitted` method.
If this is not correct, then I suggest applying debias method before `AUCMetric.fit(...)`, for example in the `calc_per_user` method. If the user calls `calc_from_fitted` or `calc_per_user_from_fitted `and passes `DebiasConfig`, then throw an exception.
@blondered @feldlime WDYT?

Reply:
`outer_merged` has a very strict and complicated structure. Simple downsampling doesn't work, there should be all ranks present in the table. When test positive needs to be removed we need to change "__test_positive" to False if rank is not Nan. Or delete the row if rank is None. This is all not obvious.
`outer_merged` in `AucFitted` instance attribute is enriched and even more complicated. We shouldn't downsample it from there. It's a pain and it's not optimal anyway.

I believe the best way will be to:
- iterate between downsampling configs in `calc_auc_metrics`
- for each config downsample interactions and then run `fit`
- for each metric calculate results from a specific fitted instance of `_AUCMetric`
It's quite similar to what you did for `MAP`

- Also we need to add downsampling in `calc_per_user` method of `_AUCMetric` right before fitting.
- And add `is_debiased` argument to `calc_from_fitted` and `calc_per_user_from_fitted`. We need to think about the naming so that we have one name for both AUC and classification metrics. I think one name will be better. So we can do smth like `is_data_debiased` or `is_input_debiased` or simple `is_debiased` everywhere.
- Remove `is_interactions_debiased` from `calc` and `calc_per_user`

@feldlime WDYT?

Reply:
Sounds fine, however I'm not really into it.  
Please cover all possible cases with the test (direct metric calculation, using `calc_metrics` with multiple metrics and same debias config, using it with different debias configs)

Reply:
Fixed this point.